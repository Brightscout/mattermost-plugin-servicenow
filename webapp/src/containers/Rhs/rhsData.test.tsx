import React from 'react';

import {shallow, ShallowWrapper} from 'enzyme';

import {configureStore} from '../../../tests/setup';
import RhsData from './rhsData';
import { Provider } from 'react-redux';

const mockSetShowAllSubscriptions = jest.fn();
const mockHandleDeleteClick = jest.fn();
const mockHandleEditSubscription = jest.fn();

const rhsDataProps = {
    error: 'mockError',
    showAllSubscriptions: true,
    setShowAllSubscriptions: mockSetShowAllSubscriptions,
    subscriptions: [],
    loadingSubscriptions: true,
    isCurrentUserSysAdmin: true,
    handleDeleteClick: mockHandleDeleteClick,
    handleEditSubscription: mockHandleEditSubscription
};

describe('Search Record Panel', () => {
    let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
    const initialState = {};
    const mockStore = configureStore();

    beforeEach(() => {
        const store = mockStore(initialState);
        component = shallow(
            <Provider store={store}>
                <RhsData {...rhsDataProps}/>
            </Provider>,
        );
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should render the rhs data body correctly', () => {
        expect(component.find(RhsData)).toHaveLength(1);
    });
});
