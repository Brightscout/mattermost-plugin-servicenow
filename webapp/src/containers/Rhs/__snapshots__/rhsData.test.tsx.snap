// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Record Panel Should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <RhsData
    error="mockError"
    handleDeleteClick={[MockFunction]}
    handleEditSubscription={[MockFunction]}
    isCurrentUserSysAdmin={true}
    loadingSubscriptions={true}
    setShowAllSubscriptions={[MockFunction]}
    showAllSubscriptions={true}
    subscriptions={Array []}
  />
</ContextProvider>
`;
